code/datums/components/uplink.dm

```diff
@@ -96,6 +97,18 @@
 	SIGNAL_HANDLER
 	SStgui.update_uis(src)
 
+/// When a new uplink is made via the syndicate beacon it locks all lockable uplinks and destroys uplinks that can not be locked
+/datum/component/uplink/proc/handle_uplink_replaced()
+	SIGNAL_HANDLER
+	if(lockable)
+		lock_uplink()
+		return
+	var/obj/item/uplink_item = parent
+	do_sparks(3,FALSE, uplink_item)
+	uplink_item.visible_message(span_warning("[uplink_item] suddenly combusts!"))
```

Comment:
1. Does this need to be shown to everyone?
2. Can this be a balloon alert instead? I don't think it's important for archival

Reply:
i made it not visible to everyone but when i tried to make it a balloon alert it runtimed because I THINK the balloon alert lasted longer than the device getting destroyed and it didnt like that